homeassistant/components/deconz/deconz_device.py

```diff
@@ -114,16 +127,17 @@ def available(self) -> bool:
         """Return True if device is available."""
         if isinstance(self._device, PydeconzScene):
             return self.gateway.available
+        assert isinstance(
+            self._device, (PydeconzGroup, PydeconzLightBase, PydeconzSensorBase)
+        )
```

Comment:
`reachable` is not part of `PydeconzDevice` so we need to be explicit on the three types here

Reply:
That seems like a mypy bug. Pyright is able to infer that `self._device` can't be `PydeconzScene` at this point.

Reply:
Should it be replaced with type:ignore? should it be guarded with if TYPE_CHECKING? or should we keep it as it is?

Reply:
Maybe `type: ignore`? At least that way we could easily detect if mypy improves and remove the comment. The `assert` might stay for a long time.

It would also help the mypy team as they use Home Assistant for their primer tests.